src/wordy_webserver.js

```diff
@@ -2,27 +2,29 @@
 
 const express = require('express');
 const bodyParser = require('body-parser');
-const helmet = require('helmet');
 // http://expressjs.com/en/advanced/best-practice-security.html
+const helmet = require('helmet');
 const response = require('./wordy_webserver_response.js');
-const auth = require('./wordy_webserver_auth.js');
+const serverAuth = require('./wordy_webserver_auth.js');
+const serverAnalytics = require('./wordy_webserver_analytics.js');
 
 class WordyWebServer {
-  constructor(storage, port, host, slackCommandToken, slackTeamId) {
+  constructor(gateway, storage, analytics, port, host, slackCommandToken, slackTeamId) {
     const webapp = express();
     webapp.use(helmet());
     webapp.use(bodyParser.urlencoded({ extended: true }));
 
     const router = express.Router();
-    router.post('/slack/command', auth.validateCommand(slackCommandToken, slackTeamId));
+    router.post('/slack/command', serverAnalytics.trackCommand(analytics));
+    router.post('/slack/command', serverAuth.validateCommand(slackCommandToken, slackTeamId));
```

Comment:
I've rarely seen this way to register middlewares for the same route. The common form is:

```js
router.post('/slack/command',
            serverAnalytics.trackCommand(analytics),
            serverAuth.validateCommand(slackCommandToken, slackTeamId));
```